### PR TITLE
Remove our fork of the numpydoc instructions in favor of the official page

### PR DESCRIPTION
**Pull request by astrofrog**
While working on #9726 I noticed that there were a number of out-of-date sections in the docs, one of which was the sentence that mentioned that we ship a fork of numpydoc in astropy. In fact we use vanilla numpydoc and don't bundle it anymore. At the time we set up our docs, the numpydoc format specification was not available on a nicely presented page so we forked them - however, the specification can now be found at https://numpydoc.readthedocs.io/en/latest/format.html which is nice and readable.

Our version has some small differences, mainly a couple of examples where we chose astropy-related variable/type names, but other than that our version is just an out of date version of the official one. So this PR is to propose simply removing our fork in favor of linking to the official one. This will also make it much simpler to stay up to date as the numpydoc package and format grow some new features.

I'm not sure if we need a changelog entry? If people are on board with this, would it make sense to include in 4.0?

### CODE REVIEW

**Comment by pllim** [comment]
I vote for adding a change log, in case someone links to that page and is confused the link is broken now.

Also, are you sure our rules are exactly the same as Numpy's? Did we resolve the "array-like" vs "array_like"?

**Comment by pllim** [comment]
Failure is real: `development_workflow.rst:315:undefined label: doc-rules`

**Comment by astrofrog** [comment]
@pllim - I think the only actual real difference in rule was ``array_like`` but that got changed back in https://github.com/astropy/astropy/pull/9684 and merged. But in any case we can take this PR as being a check for whether people are happy to fully align with the numpydoc rules now (which would make sense since we already decided to use vanilla ``numpydoc`` rather than our fork)

I'll fix the doc warning, and I can also add back an orphan page giving the link to the new page in case anyone has linked to the old doc rules from other websites.

**Comment by astrofrog** [comment]
Note to maintainers: let's wait for a few other approvals before considering whether to merge this

File: docs/development/docrules.rst [hunk 1]
```diff
@@ -1,569 +1,8 @@
-.. doctest-skip-all
-.. _doc-rules:
+:orphan:
```
Comment (bsipocz) [inline docs/development/docrules.rst]: I'm not a big fan of keeping  orphan pages. Why not delete it all?

File: CHANGES.rst [hunk 2]
```diff
@@ -335,7 +335,9 @@ astropy.wcs
 Other Changes and Additions
 ---------------------------
 
-
+- The page in the documentatin describing the astropy docstring rules has now
```
Comment (bsipocz) [inline CHANGES.rst]: I don't think we need any changelog  for this.

**Comment by bsipocz** [comment]
Release wise I would be happier with this in 4.0 than in 4.0.1.

File: docs/development/docrules.rst (hunk 1 above)
Comment (astrofrog) [reply docs/development/docrules.rst]: In case any other projects were linking to it, to avoid breaking links for now.

File: CHANGES.rst (hunk 2 above)
Comment (saimn) [reply CHANGES.rst]: typo: "documentation" (if the changelog entry stays there :))

File: docs/development/docrules.rst (hunk 1 above)
Comment (Cadair) [reply docs/development/docrules.rst]: I am pretty sure sunpy is linking to this.

**Comment by astrofrog** [comment]
I removed the changelog entry and re-milestoned to 4.0

File: docs/development/docrules.rst (hunk 1 above)
Comment (bsipocz) [reply docs/development/docrules.rst]: Then they better update their links? 🤷‍♀

**Comment by bsipocz** [comment]
There is overall happiness with 5 approvals, so I go ahead now and merge this.

Thank you @astrofrog

File: docs/development/docrules.rst (hunk 1 above)
Comment (bsipocz) [reply docs/development/docrules.rst]: I have merged this now, but my comment still stands. I don't think we should keep orphaned pages around, downstream should keep track of their links and update as needed.

**Pull request merged**